src/internals/job_lifecycle.md

```diff
@@ -153,38 +154,52 @@ enum ExitStatus {
     /// This status may be reported through the Puppet process executing within
     /// the host, and may optionally contain additional user-supplied
     /// information.
-    JobUserSuccess,
+    WorkloadFinishedSuccess,
 
     /// The host itself reports that the user-defined workload failed with an
     /// error.
     ///
     /// This status may be reported through the Puppet process executing within
     /// the host, and may optionally contain additional user-supplied
     /// information.
-    JobUserError,
+    WorkloadFinishedError,
+
+    /// The host itself reports that the user-defined workload terminated,
+    /// while indicating neither success nor failure.
+    ///
+    /// This status may be reported through the Puppet process executing within
+    /// the host, and may optionally contain additional user-supplied
+    /// information.
+    WorkloadFinishedUnknown,
 
     /// The job vanished from its supervisor, without reaching the
     /// `Terminated` execution state first.
     ///
     /// This may be due to a supervisor crash or restart. This exit status is
     /// can only be set by the Switchboard. **This exit status is final.** No
     /// subsequently reported exit status may override this status.
-    SupervisorJobDropped,
+    SupervisorDroppedJob,
+
+    /// The job timed out while dispatched.
+    ///
+    /// **This exit status is final. No subsequently reported exit status may
+    /// override this status.
+    JobTimeout,
 }
 ```
```

Comment:
We should probably add a comment here that a job does not necessarily need to have an exit status at all, but must have one at the point at which it is terminated.